pokemonterminal/platform/named_event/posix.py

```diff
@@ -0,0 +1,50 @@
+import posix_ipc
+
+from . import NamedEvent
+
+class PosixNamedEvent(NamedEvent):
+    """
+    A wrapper for named events using a named semaphore
+    """
+
+    def exists(name: str) -> bool:
+        semaphore_name = '/' + name
+        try:
+            semaphore = posix_ipc.Semaphore(semaphore_name)
+            semaphore.close()
+            return True
+        except posix_ipc.ExistentialError: # Semaphores are reconsidering their life choices
+            return False
+
+
+    def __init__(self, name: str):
+        semaphore_name = '/' + name
+        self.__semaphore = posix_ipc.Semaphore(semaphore_name, flags=posix_ipc.O_CREAT)
+
+    # NOTE this doesn't works on macOS, see http://semanchuk.com/philip/posix_ipc/#platforms
+    def is_set(self) -> bool:
+        return self.__semaphore.value > 0
+
+    def set(self):
+        self.__semaphore.release()
+
+    def clear(self):
+        try:
+            # Decrement until we reach 0 (in which case BusyError will be thrown)
+            for _ in range(self.__semaphore.value if posix_ipc.SEMAPHORE_VALUE_SUPPORTED else posix_ipc.SEMAPHORE_VALUE_MAX):
+                self.__semaphore.acquire(0)
+        except posix_ipc.BusyError:
+            return
+
+    def wait(self, timeout=None):
+        try:
+            self.__semaphore.acquire(timeout)
+        except posix_ipc.BusyError:
+            return
+
+    def name(self) -> str:
+        return self.__semaphore.name
+
+    def close(self):
+        # FIXME do we also need to close it here?
```

Comment:
I think so.
>when unlink is called, the call to unlink returns immediately but destruction of the semaphore is postponed until all processes have closed the semaphore.
So to properly destroy the semaphore it should be closed

Reply:
But then, just above, one can read `The effect of subsequent use of the semaphore by the current process is undefined.`

Reply:
but we don't use it after closing without making a new one.

Anyway, somehow the unlink here is throwing `posix_ipc.ExistentialError: No semaphore exists with the specified name`. I don't understand at all..... 😞

Reply:
Oh I see. We could unlink and *then* close it. For some reason I was fixated on doing the reverse.

Reply:
But that also means that multiple slideshow workers would crash when exiting them using the available solution (even if we do unlink only once per instance)

I think keeping the `with` and instead simply catching and silencing the error in `close()` would be better.

Reply:
I still don't really understand the point of having multiple slideshow workers, since one just takes over the other..
Anyway, testing it using `with` and catching the error works, except when I do create multiple slideshow workers I can't manage to kill them all. I can only kill the first (last?) one

Reply:
Some terminal emulators (such as Terminology), do support multiple active slideshows fine.

Only one worker being killed seems to be some unrelated bug we should investigate.